### DISCUSSION TITLE
about vue3 :class

### DISCUSSION THREAD

**Discussion by zsq44** [tailwindlabs/tailwindcss#12485] [Help]
Why is it used in Vue3 :class where some can be displayed while others cannot

**Comment by zsq44** [comment]
I mean, when I use it this way:
<div v-for="(menu,index) in data" :key="index" >
    <div class='flex flex-col shadow rounded-3xl' :class='randomStyle[index]'>
    </div>
</div>

const STYLE_BG = ['bg-red-', 'bg-blue-', 'bg-green-', 'bg-yellow-', 'bg-neutral-', 'bg-amber-', 'bg-orange-', 'bg-cyan-', 'bg-purple-', 'bg-violet-', 'bg-base-', 'bg-indigo-', 'bg-sky-', 'bg-emerald-', 'bg-fuchsia-', 'bg-stone-', 'bg-gray-', 'bg-pink-', 'bg-lime-', 'bg-rose-', 'bg-slate-', 'bg-teal-', 'bg-zinc-'];

randomStyle.value = get_random_style(STYLE_BG, '100',permission.value.length) as string[];

RanbomStyle will result in a random array, similar to ['bg cyan-100 ',' bg blue-100 '], but only a small portion being displayed

**Reply by wongjn** [reply]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

Thus, you could consider using full class names in the array, like:

```js
const STYLE_BG = ['bg-red-100', 'bg-blue-100', 'bg-green-100', 'bg-yellow-100', 'bg-neutral-100', 'bg-amber-100', 'bg-orange-100', 'bg-cyan-100', 'bg-purple-100', 'bg-violet-100', 'bg-base-100', 'bg-indigo-100', 'bg-sky-100', 'bg-emerald-100', 'bg-fuchsia-100', 'bg-stone-100', 'bg-gray-100', 'bg-pink-100', 'bg-lime-100', 'bg-rose-100', 'bg-slate-100', 'bg-teal-100', 'bg-zinc-100'];

randomStyle.value = get_random_style(STYLE_BG, permission.value.length) as string[];
```
